src/main/java/com/conveyal/gtfs/loader/ReferenceTracker.java

```diff
@@ -136,4 +150,120 @@
         }
         return errors;
     }
+
+
+    /**
+     * Work through each conditionally required check assigned to a table. First check the reference field to confirm
+     * if it meets the conditions whereby the conditional field is required. If the conditional field is required confirm
+     * that a value has been provided, if not, log an error.
+     */
+    public Set<NewGTFSError> checkConditionallyRequiredFields(Table table, Field[] fields, String[] rowData, int lineNumber) {
+        Set<NewGTFSError> errors = new HashSet<>();
+        Map<Field, ConditionalRequirement[]> fieldsToCheck = table.getConditionalRequirements();
+        for (Map.Entry<Field, ConditionalRequirement[]> entry : fieldsToCheck.entrySet()) {
+            Field referenceField = entry.getKey();
+            ConditionalRequirement[] conditionalRequirements = entry.getValue();
+            for (ConditionalRequirement check : conditionalRequirements) {
+                int refFieldIndex = Field.getFieldIndex(fields, referenceField.name);
+                String refFieldData = getValueForRow(rowData, refFieldIndex);
+                if (check.referenceCheck == ROW_COUNT_GREATER_THAN_ONE) {
+                    int conditionalFieldIndex = Field.getFieldIndex(fields, check.conditionalFieldName);
+                    String conditionalFieldData = getValueForRow(rowData, conditionalFieldIndex);
+                    if (table.name.equals("agency") &&
+                        lineNumber > 2 &&
+                        transitIds.stream().filter(transitId -> transitId.contains("agency_id")).count() != lineNumber - 1
+                    ) {
+                        String message = String.format(
+                            "%s is conditionally required when there is more than one agency.",
+                            check.conditionalFieldName
+                        );
+                        errors.add(
+                            NewGTFSError.forLine(table, lineNumber, CONDITIONALLY_REQUIRED, message)
+                        );
+                    } else if ((table.name.equals("routes")
+                        || table.name.equals("fare_attributes")) &&
+                        transitIds.stream().filter(transitId -> transitId.contains("agency_id")).count() > 1 &&
```

Comment:
This will iterate through potentially all IDs across all tables. Is there a way can separate out the `transitIds` into `agencyIds`, `routeIds`, etc?

Reply:
@evansiroky this a fair point, the `transiteIds` parameter is used by other parts of the code so I'm reluctant to make a change involving this. I'll look at using the `foreignFieldIds` parameter instead.

Reply:
Updated to use the `foreignFieldIds` parameter instead.